nixos/modules/services/networking/prosody.nix

```diff
@@ -387,6 +626,27 @@ in
         description = "Addtional path in which to look find plugins/modules";
       };
 
+      uploadHttp = mkOption {
+        description = ''
+          Configures the Prosody builtin HTTP server to handle user uploads.
+        '';
+        type = types.nullOr (types.submodule uploadHttpOpts);
+        default = null;
+        example = {
+          domain = "uploads.my-xmpp-example-host.org";
+          httpsCertificate = "/etc/letsencrypt/uploads.my-xmpp-example-host.org/cert.pem";
```

Comment:
shouldn't this be `/var/lib/acme/uploads.my-xmpp-example-host.org/cert.pem`?

Reply:
Ah! Good catch!

This example input was a WIP residue. When I started writing this PR, I first thought about using a TLS certificate per module (ie. one for the main VHost, one for the MUC, one for the HTTP upload server). Provided everything is then deployed on the same machine, it does not make any sense. I then decided to leverage the LE `extraDomains` feature to provide a multi domain single certificate.